### PR TITLE
filter system messages for push/chathead notifications

### PR DESCRIPTION
**Pull request by johnxnguyen**
The only system message notifications that were being dispatched were concerning members joining/leaving group conversations. But we only want to be notified if someone adds/deletes us from a conversation.

Additionally, we no longer need to bundle these notifications since we are not collecting participation notifications from other members (bundled notifications lead to "People were added to the group")

### CODE REVIEW

File: Source/Notifications/Push notifications/LocalNotificationDispatcher+Messages.swift [hunk 1]
```diff
@@ -131,11 +131,9 @@ extension LocalNotificationDispatcher {
     
     fileprivate func localNotificationForSystemMessage(_ message : ZMSystemMessage) -> ZMLocalNotificationForSystemMessage? {
         
-        // We might want to "bundle" notifications, e.g. member join / leave events
-        if let newNote: ZMLocalNotificationForSystemMessage = messageNotifications.copyExistingMessageNotification(message) {
```
Comment (mikeger) [inline Source/Notifications/Push notifications/LocalNotificationDispatcher+Messages.swift]: I guess bundling of the notifications is not necessary any more?

Comment (johnxnguyen) [reply Source/Notifications/Push notifications/LocalNotificationDispatcher+Messages.swift]: yeah that's right :)

File: Source/Notifications/Push notifications/LocalNotificationDispatcher+Messages.swift [hunk 2]
```diff
@@ -131,11 +131,9 @@ extension LocalNotificationDispatcher {
...
-            return newNote;
-        }
-        
+        // we only want participation messages concerning self user
+        guard message.isGroupParticipationMessageForSelf else { return nil }
```
Comment (typfel) [inline Source/Notifications/Push notifications/LocalNotificationDispatcher+Messages.swift]: Wouldn't this remove all notifications for any system message which are not `participantsAdded`, or `participantsRemoved`, like `connectionUpdate` for example. Is that what we want?

Comment (johnxnguyen) [reply Source/Notifications/Push notifications/LocalNotificationDispatcher+Messages.swift]: ah you're right. Just looking at it now, the only other system message type that produces a notification with an an `alertBody` is `.connectionRequest` (all other types put an empty string in the alert body). I'll fix the code to include this type

Comment (typfel) [reply Source/Notifications/Push notifications/LocalNotificationDispatcher+Messages.swift]: Would change the name of the function then as well.

Comment (johnxnguyen) [reply Source/Notifications/Push notifications/LocalNotificationDispatcher+Messages.swift]: when is the connection request system message sent in a conversation? At the moment, if I receive a connection request, the push notification is generated from the event, not the system message. Is the system message for something else?

**Comment by johnxnguyen** [comment]
@typfel ok so now we return early iff it is a `participantsAdded` or `participantsRemoved` system message and it is not exclusively for the self user.

**Pull request merged**